src/Microsoft.Identity.Client/ConfidentialClientApplication.cs

```diff
@@ -126,20 +126,22 @@ public sealed partial class ConfidentialClientApplication
         /// </remarks>
         /// <seealso cref="ConfidentialClientApplication"/> which
         /// enables app developers to create a confidential client application requesting tokens with the default authority.
-        [Obsolete("Use ConfidentialClientApplicationBuilder instead.")]  // TODO(migration): proper docs
+        // [Obsolete("Use ConfidentialClientApplicationBuilder instead.")]  // TODO(migration): proper docs
         public ConfidentialClientApplication(string clientId, string authority, string redirectUri,
             ClientCredential clientCredential, TokenCache userTokenCache, TokenCache appTokenCache)
             : this(ConfidentialClientApplicationBuilder
                 .Create(clientId)
                 .AddKnownAuthority(new Uri(authority), true)
                 .WithRedirectUri(redirectUri)
                 .WithClientCredential(clientCredential)
-                //.WithUserTokenCache(userTokenCache)
-                //.WithAppTokenCache(appTokenCache)
                 .BuildConfiguration())
         {
             GuardMobileFrameworks();
-            throw new NotImplementedException();
+
+            userTokenCache.SetServiceBundle(ServiceBundle);
+            UserTokenCacheInternal = userTokenCache;
+            appTokenCache.SetServiceBundle(ServiceBundle);
+            AppTokenCacheInternal = userTokenCache;
```

Comment:
appTokenCache

Reply:
exceptional catch.  thanks ;)


---
In reply to: [248454407](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/791#discussion_r248454407) [](ancestors = 248454407)